vxquery-core/src/main/java/org/apache/vxquery/serializer/XMLSerializer.java

```diff
@@ -341,6 +325,67 @@ private void printAttributeNode(PrintStream ps, TaggedValuePointable tvp) {
         }
     }
 
+    private void printObject(PrintStream ps, TaggedValuePointable tvp) {
+        ObjectPointable op = pp.takeOne(ObjectPointable.class);
+        SequencePointable seqp = pp.takeOne(SequencePointable.class);
+        VoidPointable vp = pp.takeOne(VoidPointable.class);
+        UTF8StringPointable utf8sp = pp.takeOne(UTF8StringPointable.class);
+        try {
+            tvp.getValue(op);
+            op.getKeys(tvp);
+            if (tvp.getTag() == ValueTag.SEQUENCE_TAG) {
+                tvp.getValue(seqp);
+                int len = seqp.getEntryCount();
+                ps.append('{');
+                for (int i = 0; i < len; i++) {
+                    ps.append('\"');
+                    seqp.getEntry(i, tvp);
+                    print(tvp.getByteArray(), tvp.getStartOffset(), tvp.getLength(), ps);
+                    ps.append('\"');
+                    tvp.getValue(vp);
+                    utf8sp.set(vp);
+                    ps.append(":");
+                    op.getValue(utf8sp, tvp);
+                    if (tvp.getTag() == ValueTag.XS_STRING_TAG) {
+                        ps.append('\"');
+                    }
+                    printTaggedValuePointable(ps, tvp);
+                    if (tvp.getTag() == ValueTag.XS_STRING_TAG) {
+                        ps.append('\"');
+                    }
+                    if (i != len - 1) {
+                        ps.append(',');
+                    }
+                }
+                ps.append('}');
+            } else {
+                ps.append("{\"");
+                print(tvp.getByteArray(), tvp.getStartOffset(), tvp.getLength(), ps);
+                ps.append('\"');
+                tvp.getValue(vp);
+                utf8sp.set(vp);
+                ps.append(":");
+                op.getValue(utf8sp, tvp);
+                if (tvp.getTag() == ValueTag.XS_STRING_TAG) {
```

Comment:
Here we should probably introduce a method that prints any data model instance according to the serialization specification [1] instead of just having a special case for strings.

[1] http://jsoniq.org/docs/JSONiqExtensionToXQuery/html-single/index.html#section-json-serialization